pkg/actuators/machine/machine.go

```diff
@@ -54,80 +54,245 @@ func newMachineScope(
+			}
+		}
+		if len(hostIDs) > 0 {
+			optionalPlacementPolicy, err := ovirtC.NewVMPlacementPolicyParameters().WithHostIDs(hostIDs)
+			if err != nil {
+				return errors.Wrap(err, "error creating Placement policy")
+			}
+			optionalVMParams = optionalVMParams.WithPlacementPolicy(optionalPlacementPolicy)
+		}
+	}
+
+	if ms.machineProviderSpec.Hugepages > 0 {
+		optionalVMParams = optionalVMParams.MustWithHugePages(ovirtC.VMHugePages(ms.machineProviderSpec.Hugepages))
+	}
+
+	// CREATE VM from a template
+	templateName := ms.machineProviderSpec.TemplateName
+
+	temp, err := ms.ovirtClient.GetTemplateByName(templateName)
+
+	if err != nil {
+		return errors.Wrapf(err, "error finding template name %s.", templateName)
+	}
+
+	// Handle Sparse disks and Format
+	if ms.machineProviderSpec.Sparse != nil || ms.machineProviderSpec.Format != "" {
+		tempDiskAttachment, err := ms.ovirtClient.ListTemplateDiskAttachments(temp.ID())
+		if err != nil {
+			return errors.Wrapf(err, "failed to fetch template %s disk attachments from oVirt Engine",
+				ms.machineProviderSpec.TemplateName)
+		}
+
+		diskParams := []ovirtC.OptionalVMDiskParameters{}
+		for _, diskAttachment := range tempDiskAttachment {
+			diskBuilder, _ := ovirtC.NewBuildableVMDiskParameters(diskAttachment.DiskID())
+			if ms.machineProviderSpec.Sparse != nil {
+				diskBuilder.MustWithSparse(*ms.machineProviderSpec.Sparse)
+			}
+
+			if ms.machineProviderSpec.Format != "" {
+				diskBuilder.MustWithFormat(ovirtC.ImageFormat(ms.machineProviderSpec.Format))
+			}
+
+			diskParams = append(diskParams, diskBuilder)
+		}
+		optionalVMParams = optionalVMParams.MustWithDisks(diskParams)
+	}
+
+	// Handle Disk Clone
+	if ms.machineProviderSpec.Clone != nil {
+		optionalVMParams = optionalVMParams.MustWithClone(true)
+	} else {
+		if ms.machineProviderSpec.VMType == string(ovirtsdk.VMTYPE_DESKTOP) {
+			optionalVMParams = optionalVMParams.MustWithClone(false)
+		} else {
+			optionalVMParams = optionalVMParams.MustWithClone(true)
+		}
+	}
+
+	instance, err := ms.ovirtClient.CreateVM(ovirtC.ClusterID(clusterId),
+		temp.ID(),
 		ms.machine.Name,
-		ms.machine.Labels["machine.openshift.io/cluster-api-cluster"],
-		ignition,
-		ms.machineProviderSpec)
+		optionalVMParams)
+
 	if err != nil {
 		return errors.Wrap(err, "error creating Ovirt instance")
 	}
 
 	// Wait till ready
-	err = ms.waitTillVMReachStatus(ovirtsdk.VMSTATUS_DOWN)
+	_, err = ms.ovirtClient.WaitForVMStatus(instance.ID(), ovirtC.VMStatusDown)
 	if err != nil {
 		return errors.Wrap(err, "error creating oVirt VM")
 	}
 
+	var bootableDiskAttachment ovirtC.DiskAttachment
+	newDiskSize := uint64(ms.machineProviderSpec.OSDisk.SizeGB * int64(math.Pow(2, 30)))
+	// Handle OS disk extension
+	if ms.machineProviderSpec.OSDisk != nil {
+		diskAttachments, err := instance.ListDiskAttachments()
+		if err != nil {
+			return errors.Wrapf(err, "Failed to list disk attachments for VM %s.", instance.ID())
+		}
+		for _, disk := range diskAttachments {
```

Comment:
Please rename this variable to diskAttachment, the name is confusing.

Reply:
done